src/http/ngx_http_upstream_round_robin.c

```diff
@@ -143,12 +154,18 @@ ngx_http_upstream_init_round_robin(ngx_conf_t *cf,
 
             n += server[i].naddrs;
             w += server[i].naddrs * server[i].weight;
+            g = ngx_http_upstream_gcd(g, server[i].weight);
         }
 
         if (n == 0) {
             return NGX_OK;
         }
 
+        if (g == 0) {
+            // All numbers are 0, use 1 to avoid division by 0 errors
```

Comment:
*  This comment is unnecessary that anyone can know what it is doing.
  * (BTW, for nginx codingStyle: `// xxx` -> `/* xxx */`)